services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3488,13 +3511,19 @@ private OfflinePushStatusInfo getOffLineJobStatus(
     ExecutionStatus currentReturnStatus =
         getFinalReturnStatus(sortedStatuses, childRegions, failCount, currentReturnStatusDetails);
 
+    // Do not delete parent Kafka if its part of targeted colo push to prevent concurrent pushes
     if (currentReturnStatus.isTerminal()) {
-      truncateTopicsOptionally(
-          clusterName,
-          kafkaTopic,
-          incrementalPushVersion,
-          currentReturnStatus,
-          currentReturnStatusDetails);
+      String storeName = Version.parseStoreFromKafkaTopicName(kafkaTopic);
+      int versionNum = Version.parseVersionFromKafkaTopicName(kafkaTopic);
+      boolean deferredSwap = isDeferredSwap(clusterName, storeName, versionNum);
+      if (!deferredSwap) {
```

Comment:
Is RollForward the only option after the push completes with the deferred version swap flag on?? Is there a cancel button? If so, we need to also truncate parent VT when they hit cancel.

Reply:
Right, roll forward only works if deferred swap is on. No cancel button.